## Final Project Submission

Please fill out:
* Student name: Muthuri Abigail Mwendwa
* Student pace: Full time
* Scheduled project review date/time: June 7th
* Instructor name: Maryann Mwikali

## DATA UNDERSTANDING & CLEANING

**Import Libraries**

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

#Import Numpy and the Pandas Package
import numpy as np

#Supress Warnings
import warnings
warnings.filterwarnings('ignore')

#Import Train/Test libraries
from sklearn.model_selection import train_test_split

**Read dataset**

In [2]:
df = pd.read_csv("car_insurance_claim.csv")
df.head()

ID  KIDSDRIV    BIRTH   AGE  HOMEKIDS   YOJ   INCOME PARENT1  \
0   63581743         0  16MAR39  60.0         0  11.0  $67,349      No   
1  132761049         0  21JAN56  43.0         0  11.0  $91,449      No   
2  921317019         0  18NOV51  48.0         0  11.0  $52,881      No   
3  727598473         0  05MAR64  35.0         1  10.0  $16,039      No   
4  450221861         0  05JUN48  51.0         0  14.0      NaN      No   

   HOME_VAL MSTATUS  ... CAR_TYPE RED_CAR OLDCLAIM  CLM_FREQ REVOKED MVR_PTS  \
0        $0    z_No  ...  Minivan     yes   $4,461         2      No       3   
1  $257,252    z_No  ...  Minivan     yes       $0         0      No       0   
2        $0    z_No  ...      Van     yes       $0         0      No       2   
3  $124,191     Yes  ...    z_SUV      no  $38,690         2      No       3   
4  $306,251     Yes  ...  Minivan     yes       $0         0      No       0   

   CLM_AMT CAR_AGE CLAIM_FLAG           URBANICITY  
0       $0    18.0          0  Highly Urban/ Urban  
1       $0     1.0          0  Highly Urban/ Urban  
2       $0    10.0          0  Highly Urban/ Urban  
3       $0    10.0          0  Highly Urban/ Urban  
4       $0     6.0          0  Highly Urban/ Urban  

[5 rows x 27 columns]

In [3]:
df.shape

(10302, 27)

- The dataset has 10,302 rows and 27 columns

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10302 entries, 0 to 10301
Data columns (total 27 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   ID          10302 non-null  int64  
 1   KIDSDRIV    10302 non-null  int64  
 2   BIRTH       10302 non-null  object 
 3   AGE         10295 non-null  float64
 4   HOMEKIDS    10302 non-null  int64  
 5   YOJ         9754 non-null   float64
 6   INCOME      9732 non-null   object 
 7   PARENT1     10302 non-null  object 
 8   HOME_VAL    9727 non-null   object 
 9   MSTATUS     10302 non-null  object 
 10  GENDER      10302 non-null  object 
 11  EDUCATION   10302 non-null  object 
 12  OCCUPATION  9637 non-null   object 
 13  TRAVTIME    10302 non-null  int64  
 14  CAR_USE     10302 non-null  object 
 15  BLUEBOOK    10302 non-null  object 
 16  TIF         10302 non-null  int64  
 17  CAR_TYPE    10302 non-null  object 
 18  RED_CAR     10302 non-null  object 
 19  OLDCLAIM    10302 non-nul

- My dataset contains 3 data types : int, float, object
- There are missing values in six columns 

In [5]:

# Describe the columns
column_names = [
    "ID", "KIDSDRIV", "BIRTH", "AGE", "HOMEKIDS", "YOJ", "INCOME", 
    "PARENT1", "HOME_VAL", "MSTATUS", "GENDER", "EDUCATION", 
    "OCCUPATION", "TRAVTIME", "CAR_USE", "BLUEBOOK", "TIF", "CAR_TYPE", 
    "RED_CAR", "OLDCLAIM", "CLM_FREQ", "REVOKED", "MVR_PTS", "CLM_AMT", 
    "CAR_AGE", "CLAIM_FLAG", "URBANICITY"
]

column_descriptions = [
    "Unique identifier for each customer.", "Number of kids the customer has who are of driving age.", 
    "Date of birth of the customer.", "Age of the customer.", 
    "Number of kids the customer has at home.", "Years on the job (number of years the customer has been employed).", 
    "Income of the customer.", "Indicates whether the customer is a single parent (Yes/No).", 
    "Value of the customer's home.", "Marital status of the customer (Yes/No).", 
    "Gender of the customer (M/F).", "Education level of the customer.", 
    "Occupation of the customer.", "Daily travel time to work.", 
    "Purpose of car usage (Private/Commercial).", "Value of the customer's car.", 
    "Time in force (Years).", "Type of car.", "Indicates whether the customer owns a red car (yes/no).", 
    "Total amount of previous claims.", "Number of claims reported.", 
    "Indicates whether the customer's driver's license has been revoked (Yes/No).", 
    "Number of motor vehicle record points.", "Possible future claims", 
    "Age of the customer's car.", "Indicates whether a claim was filed (1/0).", 
    "Urbanicity type ((Highly Urban/Urban)/(Highly Rural/Rural))."
]

# Create a DataFrame
data = {'Column Name': column_names, 'Description': column_descriptions}
desc_df = pd.DataFrame(data)

desc_df

Column Name                                        Description
0           ID               Unique identifier for each customer.
1     KIDSDRIV  Number of kids the customer has who are of dri...
2        BIRTH                     Date of birth of the customer.
3          AGE                               Age of the customer.
4     HOMEKIDS           Number of kids the customer has at home.
5          YOJ  Years on the job (number of years the customer...
6       INCOME                            Income of the customer.
7      PARENT1  Indicates whether the customer is a single par...
8     HOME_VAL                      Value of the customer's home.
9      MSTATUS           Marital status of the customer (Yes/No).
10      GENDER                      Gender of the customer (M/F).
11   EDUCATION                   Education level of the customer.
12  OCCUPATION                        Occupation of the customer.
13    TRAVTIME                         Daily travel time to work.
14     CAR_USE         Purpose of car usage (Private/Commercial).
15    BLUEBOOK                       Value of the customer's car.
16         TIF                             Time in force (Years).
17    CAR_TYPE                                       Type of car.
18     RED_CAR  Indicates whether the customer owns a red car ...
19    OLDCLAIM                   Total amount of previous claims.
20    CLM_FREQ                         Number of claims reported.
21     REVOKED  Indicates whether the customer's driver's lice...
22     MVR_PTS             Number of motor vehicle record points.
23     CLM_AMT                             Possible future claims
24     CAR_AGE                         Age of the customer's car.
25  CLAIM_FLAG         Indicates whether a claim was filed (1/0).
26  URBANICITY  Urbanicity type ((Highly Urban/Urban)/(Highly ...

### **Cleaning Dataset**

- Convert income and home value to numeric, removing the dollar sign and commas

In [6]:
df['INCOME'] = df['INCOME'].replace('[\$,]', '', regex=True).astype(float)

In [7]:
df['HOME_VAL'] = df['HOME_VAL'].replace('[\$,]', '', regex=True).astype(float)

- Convert old claim and claim amount to numeric, removing the dollar sign and commas

In [8]:
df['OLDCLAIM'] = df['OLDCLAIM'].replace('[\$,]', '', regex=True).astype(float)

In [9]:
df['CLM_AMT'] = df['CLM_AMT'].replace('[\$,]', '', regex=True).astype(float)

- Check for missing values

In [10]:
df.isna().sum()

ID              0
KIDSDRIV        0
BIRTH           0
AGE             7
HOMEKIDS        0
YOJ           548
INCOME        570
PARENT1         0
HOME_VAL      575
MSTATUS         0
GENDER          0
EDUCATION       0
OCCUPATION    665
TRAVTIME        0
CAR_USE         0
BLUEBOOK        0
TIF             0
CAR_TYPE        0
RED_CAR         0
OLDCLAIM        0
CLM_FREQ        0
REVOKED         0
MVR_PTS         0
CLM_AMT         0
CAR_AGE       639
CLAIM_FLAG      0
URBANICITY      0
dtype: int64

- Drop missing values

In [11]:
df.dropna(inplace = True)

- Check for duplicates

In [12]:
duplicates_with_rows = df[df.duplicated()]
duplicates_with_rows

ID  KIDSDRIV    BIRTH   AGE  HOMEKIDS   YOJ   INCOME PARENT1  \
9801  279799481         0  21FEB60  39.0         0  14.0  93077.0      No   

      HOME_VAL MSTATUS  ... CAR_TYPE RED_CAR OLDCLAIM  CLM_FREQ REVOKED  \
9801  244764.0     Yes  ...  Minivan     yes      0.0         0      No   

     MVR_PTS  CLM_AMT CAR_AGE CLAIM_FLAG             URBANICITY  
9801       0      0.0     1.0          0  z_Highly Rural/ Rural  

[1 rows x 27 columns]

In [13]:
# Remove Duplicates
df = df.drop_duplicates()

- Check if dataset has any more missing values

In [14]:
df.isna().sum()

ID            0
KIDSDRIV      0
BIRTH         0
AGE           0
HOMEKIDS      0
YOJ           0
INCOME        0
PARENT1       0
HOME_VAL      0
MSTATUS       0
GENDER        0
EDUCATION     0
OCCUPATION    0
TRAVTIME      0
CAR_USE       0
BLUEBOOK      0
TIF           0
CAR_TYPE      0
RED_CAR       0
OLDCLAIM      0
CLM_FREQ      0
REVOKED       0
MVR_PTS       0
CLM_AMT       0
CAR_AGE       0
CLAIM_FLAG    0
URBANICITY    0
dtype: int64

In [15]:
df.shape

(7656, 27)

 - After cleaning, the dataset has 7656 rows and 27 columns